internal/providers/terraform/azure/mssql_database.go

```diff
@@ -0,0 +1,270 @@
+			HourlyQuantity: replicaCount,
+			ProductFilter: &schema.ProductFilter{
+				VendorName:    strPtr("azure"),
+				Region:        strPtr(region),
+				Service:       strPtr(serviceName),
+				ProductFamily: strPtr("Databases"),
+				AttributeFilters: []*schema.AttributeFilter{
+					{Key: "productName", ValueRegex: strPtr(productNameRegex)},
+					{Key: "skuName", Value: strPtr(skuName)},
+				},
+			},
+			PriceFilter: &schema.PriceFilter{
+				PurchaseOption: strPtr("Consumption"),
+			},
+		})
+	}
+
+	if tier == "GP" || tier == "HS" || tier == "BC" {
+		var licenseType string
+		if d.Get("license_type").Type != gjson.Null {
+			licenseType = d.Get("license_type").String()
+			if licenseType == "LicenseIncluded" {
+				costComponents = append(costComponents, sqlLicenseCostComponent(region, cores, serviceName, tier))
+			}
+		}
+	}
+
+	storageGb := decimalPtr(decimal.NewFromInt(5))
+	if d.Get("max_size_gb").Type != gjson.Null {
+		storageGb = decimalPtr(decimal.NewFromInt(d.Get("max_size_gb").Int()))
+	}
+	costComponents = append(costComponents, mssqlStorageComponent(storageGb, region, serviceName, tier, zoneRedundant))
+
+	var retention *decimal.Decimal
+	if tier != "HS" {
+		if u != nil && u.Get("monthly_long_term_retention_storage_gb").Exists() {
+			retention = decimalPtr(decimal.NewFromInt(u.Get("monthly_long_term_retention_storage_gb").Int()))
+		}
+		costComponents = append(costComponents, &schema.CostComponent{
+			Name:            "Long-term retention",
+			Unit:            "GB-months",
+			UnitMultiplier:  1,
+			MonthlyQuantity: retention,
+			ProductFilter: &schema.ProductFilter{
+				VendorName:    strPtr("azure"),
+				Region:        strPtr(region),
+				Service:       strPtr(serviceName),
+				ProductFamily: strPtr("Databases"),
+				AttributeFilters: []*schema.AttributeFilter{
+					{Key: "productName", ValueRegex: strPtr("/LTR Backup Storage/")},
+					{Key: "skuName", Value: strPtr("Backup RA-GRS")},
+					{Key: "meterName", Value: strPtr("RA-GRS Data Stored")},
+				},
+			},
+			PriceFilter: &schema.PriceFilter{
+				PurchaseOption: strPtr("Consumption"),
+			},
+		})
+	}
+
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: costComponents,
+	}
+}
+
+func mssqlSkuSplit(address, sku string) (string, string, string, error) {
+	if s := strings.Split(sku, "_"); len(s) == 4 && strings.HasPrefix(sku, "GP_S_Gen") {
+		return s[0] + s[1], s[2], s[3], nil
+	}
+	if s := strings.Split(sku, "_"); len(s) == 3 {
+		if strings.HasPrefix(sku, "HS_Gen") ||
+			strings.HasPrefix(sku, "HS_DC") ||
+			strings.HasPrefix(sku, "GP_Gen") ||
+			strings.HasPrefix(sku, "GP_Fsv2") ||
+			strings.HasPrefix(sku, "GP_DC") ||
+			strings.HasPrefix(sku, "BC_Gen") ||
+			strings.HasPrefix(sku, "BC_M") ||
+			strings.HasPrefix(sku, "BC_DC") ||
+			strings.HasPrefix(sku, "BC_DC") {
+			return s[0], s[1], s[2], nil
+		}
+	}
+	return "", "", "", errors.Errorf("Unrecognised MSSQL SKU format for resource %s: %s", address, sku)
+}
```

Comment:
This wouldn't check that the combinations of tier+family+core are valid, but I think that's okay

Reply:
As a side note (and this was probably my fault). We should use `Unrecognized` instead of `Unrecognised`, since we should stick to US English spelling.

Reply:
Yeah, this is nice idea, looking good. I'll use it